integration-cli/docker_utils.go

```diff
@@ -396,6 +396,55 @@ func deleteAllContainers() error {
 	return nil
 }
 
+var protectedImages = map[string]struct{}{}
+
+func init() {
+	out, err := exec.Command(dockerBinary, "images").CombinedOutput()
+	if err != nil {
+		panic(err)
+	}
+	lines := strings.Split(string(out), "\n")[1:]
+	for _, l := range lines {
+		if l == "" {
+			continue
+		}
+		fields := strings.Fields(l)
+		imgTag := fields[0] + ":" + fields[1]
+		protectedImages[imgTag] = struct{}{}
```

Comment:
It could have any number of images, including but not at all limited to
frozen images because you can run the test suite against an arbitrary
daemon.  I think the code as written is great, and offers protection for
the most important parts (named images that already exist), but it has fun
behavior in the extreme edge case of pre-existing dangling images, so I'm
just trying to work through that verbally so we can decide whether the
current "preserve them all if any exist" side effect is how we want to
handle that. :+1: